.github/workflows/e2e-test.yml

```diff
@@ -3,28 +3,44 @@ on:
   push:
     branches:
       - main
+    paths:
+      - '**.go'
+      - 'Makefile'
+      - 'go.md'
+      - 'go.sum'
+      - '.github/workflows/*.yml'
+      - 'test/**'
+      - 'hack/e2e/**'
+      - 'hack/terraform/**'
 jobs:
-  e2e-on-linux:
-    runs-on: ubuntu-latest
+  e2e-test:
+    if: github.repository == 'devstream-io/devstream'
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        go: [1.17.x]
     permissions:
       id-token: write
       contents: write
+    runs-on: ${{ matrix.os }}
+    name: e2e-test-${{ matrix.os }}
     steps:
       - run: echo "🐧 This job is now running on a ${{ runner.os }}-${{ runner.arch }} server hosted by GitHub!"
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Setup Golang env
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: ${{ matrix.go }}
       - name: Cache
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           path: |
-            ~/.cache/go-build
             ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ matrix.go }}-${{ hashFiles('**/go.sum') }}
           restore-keys: |
+            ${{ runner.os }}-go-${{ matrix.go }}-
             ${{ runner.os }}-go-
       - name: Build
         run: make build -j8
```

Comment:
If we remove the `push to main` workflow, the ut will never run at main branch. Should we add the ut step here?

Reply:
The e2e test is triggered when merging into the main branch

Reply:
> If we remove the `push to main` workflow, the ut will never run at main branch. Should we add the ut step here?

`pull request` will trigger ut, how about having `main.yml` support push events?

Reply:
Of course, after adding the `push events trigger`, this logic is complete. It's just that I see the simplification you have done. I guess it is to reduce the workflow on the main branch updated, so it seems easier to add `ut` to e2e.yml; if we add `push` directly to main.yml, it seems that `main` and `e2e` cannot avoid repeated `make build`.